engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/TaskRestService.java

```diff
@@ -29,14 +29,15 @@
 import javax.ws.rs.core.Request;
 import javax.ws.rs.core.UriInfo;
 import java.util.List;
+import java.util.Optional;
 
 @Produces(MediaType.APPLICATION_JSON)
 public interface TaskRestService {
 
   public static final String PATH = "/task";
 
   @Path("/{id}")
-  TaskResource getTask(@PathParam("id") String id);
+  TaskResource getTask(@PathParam("id") String id, @QueryParam("withCommentAttachmentInfo") Optional<Boolean> withCommentAttachmentInfo);
```

Comment:
Note to myself: check CI and if there are cons against using Optional.

Reply:
@sumankumarpani, the `engine-rest` changes are breaking the tests for one of our supported environments. I am checking how is best to adjust the new parameter and preserve backwards compatibility. I will contact you further with my findings.

If you want to see the failures for your self, you can execute the following (with JDK 11):
`mvn clean install -Pwildfly-compatibility -f engine-rest/engine-rest/pom.xml`

Reply:
@yanavasileva I am seeing this error, are you seeing the same.
Response body doesn't match expectation.
Expected: is ""
  Actual: {"type":"RuntimeException","message":"RESTEASY003875: Unable to find a constructor that takes a String param or a valueOf() or fromString() method for javax.ws.rs.QueryParam(\"withCommentAttachmentInfo\") on public org.camunda.bpm.engine.rest.sub.task.TaskResource org.camunda.bpm.engine.rest.impl.TaskRestServiceImpl.getTask(java.lang.String,java.util.Optional) for basetype: java.util.Optional","code":null}

[ERROR]   TaskVariableRestResourceInteractionTest.testGetObjectVariables:162 5 expectations failed.
Expected status code <200> but was <500>.

I couldn't find the exact reason yet, will update here if I get any clue.
@yanavasileva I am seeing this error, are you seeing the same.
Response body doesn't match expectation.
Expected: is ""
  Actual: {"type":"RuntimeException","message":"RESTEASY003875: Unable to find a constructor that takes a String param or a valueOf() or fromString() method for javax.ws.rs.QueryParam(\"withCommentAttachmentInfo\") on public org.camunda.bpm.engine.rest.sub.task.TaskResource org.camunda.bpm.engine.rest.impl.TaskRestServiceImpl.getTask(java.lang.String,java.util.Optional) for basetype: java.util.Optional","code":null}

[ERROR]   TaskVariableRestResourceInteractionTest.testGetObjectVariables:162 5 expectations failed.
Expected status code <200> but was <500>.

I couldn't find the exact reason yet, will update here if I get any clue. Meanwhile let me know if you recall handling similar in the past.

Reply:
@yanavasileva  I modified the the param  Optional<Boolean> withCommentAttachmentInfo to Boolean withCommentAttachmentInfo and given it a default value false, to make it work. Let me know if we can proceed with the solution.

Reply:
It seems like a better fit. We use similar approach in other places too.
Please adjust the code, I will run then the tests again.

Reply:
@yanavasileva  I have pushed changes , please take a look.